x/interchainquery/keeper/keeper.go

```diff
@@ -78,6 +78,9 @@ func (k *Keeper) RetryICQRequest(ctx sdk.Context, query types.Query) error {
 		return err
 	}
 
+	// TODO [LSM]: Investigate why we need to force a unique ID when retrying queries
```

Comment:
Could you expand on this issue?

Reply:
For some reason, it seemed like the query events weren't getting emitted when I tried to resubmit with the same query ID. I'm going to reach out to Andrew - I assume there's an assumption the relayer makes